src/realizations/catchment/Bmi_Multi_Formulation.cpp

```diff
@@ -199,7 +215,27 @@ void Bmi_Multi_Formulation::create_multi_formulation(geojson::PropertyMap proper
             LOG("Deprecated output_header_fields item found in realization file ignored.", LogLevel::WARNING);
         }
     }
-    
+
+    //check if units have not been specified. If not, default to native units.
+    std::string blank_string = "";
+    auto &names = get_output_variable_names();
+    if(out_units.size() == 0){
+        out_units.resize(names.size());
+        for (int i = 0; i < names.size(); ++i) {
+            out_units[i] = get_bmi_native_units(names[i]);
+        }
+        set_output_variable_units(out_units);
+    }
+    else if(std::find(out_units.begin(), out_units.end(), blank_string) != out_units.end()){
+        //this condition is when the user has not specified units for an output variable.
+        for (int i = 0; i < names.size(); ++i) {
+            if (out_units[i] == blank_string){
+                out_units[i] = get_bmi_native_units(names[i]);
+            }
+        }
+        set_output_variable_units(out_units);
+    }
```

Comment:
The repeated logic here can be greatly simplified:

```suggestion
    auto &names = get_output_variable_names();
    if(out_units.size() == 0){
        out_units.resize(names.size(), blank_string);
    }

    for (int i = 0; i < names.size(); ++i) {
        if (out_units[i] == blank_string){
            out_units[i] = get_bmi_native_units(names[i]);
        }
    }
    set_output_variable_units(out_units);
```